### PR TITLE
docs(carbon-react): add README and initial sass config docs

### LINKED ISSUE #8558: Add initial README for `@carbon/react`
This README will be what is made available under the preview package that we will ship for the beta release. It should include the following set of information:

- [ ] How to install the package
- [ ] How to use styles
- [ ] How to use icons
- [ ] How to bring in components

### PR DESCRIPTION
**Pull request by joshblack**
Closes #8558

This PR adds in our initial README and sass doc for our beta release next week. The goal with this work is to provide the minimal documentation needed for teams to get started with:

- Including components
- Including icons
- Including styles

It's also a chance for our team to see the new package architecture and see how it fits in with our plans moving forward 👀 

#### Changelog

**New**

- Add README for `@carbon/react`
- Add initial docs for sass configuration for `@carbon/react`

**Changed**

- Add sass as a peer dependency for `@carbon/react`

**Removed**

#### Testing / Reviewing

- Read through the README / Sass configuration docs to see if they make sense
- Take a look at the new package architecture (one package versus several) for bringing in parts of our design system and see if it makes sense to you / is it the right call for our team

### CODE REVIEW

File: packages/carbon-react/README.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+**Note: this package is unstable and will change before it's 1.0 release**
+
+You can use the `@carbon/react` to bring in components, icons, and styles from
+the Carbon Design System.
+
+To include a component, you can import it:
+
+```jsx
+import { Button } from '@carbon/icons-react';
```
Comment (tay1orjones) [inline packages/carbon-react/README.md]:
```suggestion
import { Button } from '@carbon/react';
```

File: packages/carbon-react/README.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+function MyComponent() {
+  return <Button>Example usage</Button>;
+}
+```
+
+To include an icon, use the `@carbon/icons-react/icons` entrypoint and import it
```
Comment (tay1orjones) [inline packages/carbon-react/README.md]:
```suggestion
To include an icon, use the `@carbon/react/icons` entrypoint and import it
```

File: packages/carbon-react/README.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+the same way you would import a component:
+
+```jsx
+import { Add } from '@carbon/icons-react/icons';
```
Comment (tay1orjones) [inline packages/carbon-react/README.md]:
```suggestion
import { Add } from '@carbon/react/icons';
```

File: packages/carbon-react/README.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+```jsx
+@use '@carbon/react';
+```
+
+Note: you will need to configure Sass in order to correctly find modules in your
+`node_modules`. Follow [this guide](./docs/sass.md#configuration) to make sure
+your project is setup correctly.
+
+For more information about each component, check out our
+[Storybook](https://react.carbondesignsystem.com). For guides on getting
```
Comment (tay1orjones) [inline packages/carbon-react/README.md]: I know this PR is focused on docs, but curious if the new package should ship it's own storybook for the beta?

File: packages/carbon-react/README.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+## :books: Documentation
+
+- [Storybook](https://react.carbondesignsystem.com)
+- [Getting started](https://www.carbondesignsystem.com/get-started/develop/react)
+- [🏃‍♀️ Migration Guides](./docs/migration)
+  - [v6 to v7](./docs/migration/migrate-to-7.x.md)
+
+## 📚 Examples
+
+If you're looking for more examples on how to use `carbon-components-react`, we
```
Comment (tay1orjones) [inline packages/carbon-react/README.md]:
```suggestion
If you're looking for more examples on how to use `@carbon/react`, we
```

File: packages/carbon-react/docs/sass.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+options passed into `sassOptions`:
+
+```js
+{
+  loader: 'sass-loader',
+  options: {
+    sassOptions: {
+      includePaths: ['node_modules'],
+    },
+  },
```
Comment (tay1orjones) [inline packages/carbon-react/docs/sass.md:77]: I know `sass-loader` prefers the dart sass implementation by default, but should we call that out here? and or just recommend configuring the `implementation` option?

```suggestion
  options: {
    implementation: require("sass"),
    sassOptions: {
      includePaths: ['node_modules'],
    },
  },
```

File: packages/carbon-react/package.json [hunk 7]
```diff
@@ -35,7 +35,8 @@
   },
   "peerDependencies": {
     "react": "^16.8.6 || ^17.0.1",
-    "react-dom": "^16.8.6 || ^17.0.1"
+    "react-dom": "^16.8.6 || ^17.0.1",
+    "sass": ">= 1.32"
```
Comment (tay1orjones) [inline packages/carbon-react/package.json]: Would we want a caret range instead so a new major wouldn't be an acceptable peer dep?

**Review by tay1orjones — COMMENTED** [review]
Just a couple things that stuck out, but overall looks great to me!

File: packages/carbon-react/README.md (hunk 4 above)
Comment (joshblack) [reply packages/carbon-react/README.md]: @tay1orjones yes! Definitely, it should. It's one of the task items this sprint that I'm hoping to finish up this week 🤞

File: packages/carbon-react/docs/sass.md (hunk 6 above)
Comment (joshblack) [reply packages/carbon-react/docs/sass.md:77]: Maybe we could add a note like "if the styles aren't working, you might be using node-sass in your project" and then use this as a way to tell if that's the case?

File: packages/carbon-react/package.json (hunk 7 above)
Comment (joshblack) [reply packages/carbon-react/package.json]: Hell yeah, that makes a lot of sense.

**Review by tay1orjones — APPROVED** [review]
Awesome! 🎉

File: packages/carbon-react/README.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+```
+
+To include an icon, use the `@carbon/react/icons` entrypoint and import it the
+same way you would import a component:
+
+```jsx
+import { Add } from '@carbon/react/icons';
+
+function MyComponent() {
+  return <Add />;
```
Comment (tw15egan) [inline packages/carbon-react/README.md:45]: Will users be able to specify a size? Something like `<Add size={32} />`?

File: packages/carbon-react/README.md [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+
+```jsx
+@use '@carbon/react';
+```
+
+Note: you will need to configure Sass in order to correctly find modules in your
+`node_modules`. Follow [this guide](./docs/sass.md#configuration) to make sure
+your project is setup correctly.
+
+## :books: Documentation
```
Comment (tw15egan) [inline packages/carbon-react/README.md:59]: Seems like this emoji didn't come through

**Review by tw15egan — APPROVED** [review]
Looks great! Just a few comments

File: packages/carbon-react/README.md (hunk 8 above)
Comment (joshblack) [reply packages/carbon-react/README.md:45]: @tw15egan yes! Definitely. Will try and get some explicit docs for it, but for now we have: https://github.com/carbon-design-system/carbon/blob/main/docs/migration/11.x-carbon-icons-react.md#overview to cover size albeit indirectly

Comment (tw15egan) [reply packages/carbon-react/README.md:45]: Perfect!

File: packages/carbon-react/README.md (hunk 9 above)
Comment (joshblack) [reply packages/carbon-react/README.md:59]: Oh, it's a GH-flavored emoji, it seems. It should render in the markdown 👀

Comment (tw15egan) [reply packages/carbon-react/README.md:59]: Weird, not sure why the other one worked locally 🤷🏻‍♂️

**Pull request merged**